### PR TITLE
Add basic postcondition checker for errors

### LINKED ISSUE #1095: Design and prototype common storage interface
Do enough prototyping so future tasks can be created and estimated.

- [x] Add basic contract interface.
- [x] Add basic contract validation for common errors.
- [x] Add basic observability
- [x] Re-think internal error handling in storages

### PR DESCRIPTION
**Pull request by AlekSi**
# Description

Closes #1095.

## Readiness checklist

* [x] I added/updated unit tests.
* [ ] I added/updated integration/compatibility tests.
* [x] I added/updated comments and checked rendering.
* [x] I made spot refactorings.
* [ ] I updated user documentation.
* [x] I ran `task all`, and it passed.
* [x] I ensured that PR title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

**Review by w84thesun — APPROVED** [review]
👍

File: internal/handlers/pg/pgdb/databases.go [hunk 1]
```diff
@@ -89,25 +90,27 @@ func CreateDatabaseIfNotExists(ctx context.Context, tx pgx.Tx, db string) error
...
 
 // DropDatabase drops FerretDB database.
-//
-// It returns ErrSchemaNotExist if schema does not exist.
-func DropDatabase(ctx context.Context, tx pgx.Tx, db string) error {
-	_, err := tx.Exec(ctx, `DROP SCHEMA `+pgx.Identifier{db}.Sanitize()+` CASCADE`)
-	if err == nil {
-		return nil
+func DropDatabase(ctx context.Context, tx pgx.Tx, db string) (err error) {
+	defer contract.EnsureError(err, ErrSchemaNotExist) // if schema does not exist
```
Comment (rumyantseva) [inline internal/handlers/pg/pgdb/databases.go:94]: This approach looks helpful! We can also remove `case errors.Is(err, pgdb.ErrSchemaNotExist):` from `MsgDropDatabase`.

Comment (AlekSi) [reply internal/handlers/pg/pgdb/databases.go:94]: We can't do that because contracts are not enforced in non-debug builds

Comment (rumyantseva) [reply internal/handlers/pg/pgdb/databases.go:94]: Ah, ok! I misunderstood the "It does nothing in non-debug builds" part. The part I didn't get is that the original error is kept.

**Review by rumyantseva — APPROVED** [review]

**Pull request merged**